superset-frontend/src/features/alerts/AlertReportModal.tsx

```diff
@@ -1001,30 +1047,102 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
     return hasInfo;
   };
 
-  const validate = () => {
+  const validateGeneralSection = () => {
+    const errors = [];
+    if (!currentAlert?.name?.length) {
+      errors.push('name');
+    }
+    if (!currentAlert?.owners?.length) {
+      errors.push('owners');
+    }
+    if (errors.length) {
+      updateValidationStatus(Sections.GENERAL, false, errors);
+    } else {
+      updateValidationStatus(Sections.GENERAL, true);
+    }
+  };
+  const validateContentSection = () => {
+    const errors = [];
     if (
-      currentAlert?.name?.length &&
-      currentAlert?.owners?.length &&
-      currentAlert?.crontab?.length &&
-      currentAlert?.working_timeout !== undefined &&
-      ((contentType === 'dashboard' && !!currentAlert?.dashboard) ||
-        (contentType === 'chart' && !!currentAlert?.chart)) &&
-      checkNotificationSettings()
+      !(
+        (contentType === 'dashboard' && !!currentAlert?.dashboard) ||
+        (contentType === 'chart' && !!currentAlert?.chart)
+      )
     ) {
-      if (isReport) {
-        setDisableSave(false);
-      } else if (
-        !!currentAlert.database &&
-        currentAlert.sql?.length &&
-        (conditionNotNull || !!currentAlert.validator_config_json?.op) &&
+      errors.push('content type');
+    }
+    if (errors.length) {
+      updateValidationStatus(Sections.CONTENT, false, errors);
+    } else {
+      updateValidationStatus(Sections.CONTENT, true);
+    }
+  };
+  const validateAlertSection = () => {
+    const errors = [];
+    if (!currentAlert?.database) {
+      errors.push('database');
```

Comment:
I see these being references in a bunch of places as strings and logics being tied to them. I believe this is a good opportunity to use an Enum for these rather than plain strings

Reply:
This and other enum comments above have been incorporated into the TRANSLATIONS object.